scripts/lib/webServer.ts

```diff
@@ -14,6 +14,17 @@ import { $ } from "zx";
 
 export async function startWebServer(directory: string, listenPort: number) {
   $`python3 -m http.server ${listenPort} -d ${directory} -b ::1 &`;
+
+  // Wait until the server is up and able to listen to the requests
+  while (true) {
```

Comment:
Instead, a better approach is to use exponential backoff. It avoids sending too many requests all at once. Whereas right now, once a request fails, you immediately are sending another. This library looks good: https://www.npmjs.com/package/exponential-backoff

But you can also avoid using that library and keep this simpler. Do something like this strategy:

1. No matter what, wait one second
2. Check if valid
3. Wait 3 more seconds
4. Check if valid. Else exit

The setup is simple enough that you can make it explicit and Keep It Simple. I'd recommend this approach.

Reply:
Thanks for the comment! I also thought it was better to keep it simple, and given that the time needed is close to 500 ms, I used the while loop. It's a good idea to wait before the first validation and then check if the server is listening, so I followed the strategy you said in the comment :+1: The times I used are 500 ms in the first wait and 1 second in the second wait (We should perform 1 or 2 fetches at most).

Reply:
I'm not following how this while loop is safe. If the first time we do the `try` block fails, then we enter the `catch` block and wait 1 second. Then we try again. That can keep happening in an infinite loop, right? There is no code to say at a max, try 1-2 times.

I think you can remove the `while` loop and make your 2 attempts more explicit. It's fine in this case to repeat a little bit of code.